node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -3593,21 +3593,25 @@ mod tests {
     fn standard_gossip_containing_unfamiliar_node_addrs_leads_to_them_being_ignored() {
         /*
 
+        <---- Databases before the gossip ---->
+
         Destination Database (Root) ==>
 
-            Root
-            / \
-           A---B---D
+          Root
+         /  |
+        B---A
+        |
+        D
 
         Source Database (A) ==>
-        - Source has updated version of A and B.
-        - C will not reveal IP, but B, E and F will do.
 
-           Root
+          Root
          /  |
         B---A---E
         |   |
         C   F
+
+        <------------------------------------->
```

Comment:
I still think this test has been long-time wrong, not in a critical way, but those flags for reveal-node-addr should be right only as: 
```
let gossip = GossipBuilder::new(&src_db)
            .node(node_a.public_key(), true)
            .node(node_b.public_key(), true)
            .node(node_c.public_key(), false)
            .node(node_e.public_key(), false)
            .node(node_f.public_key(), false)
            .build();
```